Code/Framework/AzCore/AzCore/Serialization/EditContext.h

```diff
@@ -819,16 +819,15 @@ namespace AZ
         {
             AZ_TYPE_INFO(EnumConstant, "{4CDFEE70-7271-4B27-833B-F8F72AA64C40}");
 
-            typedef typename AZStd::RemoveEnum<EnumType>::type UnderlyingType;
-
             EnumConstant() {}
-            EnumConstant(EnumType first, const char* description)
+            EnumConstant(EnumType first, AZStd::string_view description)
+                : m_value(aznumeric_cast<AZ::u64>(first))
```

Comment:
This should use `static_cast`
A numeric_cast is invalid in the case when an enum with a negative is used

Reply:
Strange, I did try with a negative case and `numeric_cast` didn't complain - I don't even see anything logged to stdout or stderr. I'll still make the change, it makes sense to eschew it for `static_cast` anyways, but I would have thought it would yell at me.

Reply:
That is because numeric_cast is only enabled in UnitTest currently.